### PR TITLE
fixed bug with negative constraints, added integration test

### PR DESCRIPTION
**Pull request by mjpost**
Check for negative constraints didn't work when there are none. Added integration test that would have caught this.

#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [ ] Were system tests modified? If so did you run these at least 5 times to account for the variation across runs?
- [ ] System tests pass (`pytest test/system`)
- [x] Passed code style checking (`./style-check.sh`)
- [ ] You have considered writing a test
- [x] Updated major/minor version in `sockeye/__init__.py`. Major version bump if this is a backwards incompatible change.
- [ ] Updated CHANGELOG.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mjpost** [comment]
The failing tests all have to do with `pylint==2.0.1` which is now on travis. Perhaps related to #478.

File: sockeye/inference.py [hunk 1]
```diff
@@ -690,22 +690,29 @@ def make_input_from_json_string(sentence_id: int, json_string: str) -> Translato
...
-            constraints = None
 
-        # List of phrases to prevent from occuring in the output
-        avoid_list = jobj.get(C.JSON_AVOID_KEY)
+        # If there is overlap between positive and negative constraints, assume the user wanted
+        # the words, and so remove them from the avoid_list (negative constraints)
+        if constraints is not None and avoid_list is not None:
+            overlap = set(constraints).intersection(set(avoid_list))
+            if len(overlap) > 0:
+                new_avoid_list = []
```
Comment (tdomhan) [inline sockeye/inference.py]: shorter:
```
new_avoid_list = list(avoid_set.difference(overlap))
```

File: test/common.py [hunk 2]
```diff
@@ -204,7 +205,11 @@ def tmp_digits_dataset(prefix: str,
                         start_pos = 0
                         end_pos = min(target_len, 3)
                         constraint = ' '.join(target_words[start_pos:end_pos])
-                        new_source['constraints'] = [constraint]
+                        # Alternate between positive and negative constraints
```
Comment (tdomhan) [inline test/common.py]: should we add the case of both here as well?

File: sockeye/inference.py (hunk 1 above)
Comment (mjpost) [reply sockeye/inference.py]: Ah, much nicer.

File: test/common.py (hunk 2 above)
Comment (mjpost) [reply test/common.py]: I think we should keep them separate, just because I don't want to always test both.

Comment (tdomhan) [reply test/common.py]: ok, sure. We could create a third branch add either constraints, avoid, or both.

**Comment by tdomhan** [comment]
let's merge after travis.

**Pull request merged**